metacat-connector-druid/src/main/java/com/netflix/metacat/connector/druid/converter/Interval.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.connector.druid.converter;
+
+import lombok.Data;
+
+import java.util.Date;
+
+/**
+ * Interval.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Data
+public class Interval implements Comparable<Interval> {
```

Comment:
I think you should look at using the [Joda Time interval](http://www.joda.org/joda-time/apidocs/org/joda/time/Interval.html) class rather than rolling your own. At the very least try to avoid using the Java `Date` class and use Java 8 [`Instant`](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html)